github-pr-dashboard/src/components/Container/Container.js

```diff
@@ -0,0 +1,116 @@
+//This component renders the columns and passes the filtered data to next component(List)
+
+//React's and apollo's dependencies
+import { React, useEffect, useState } from "react";
+import { useQuery } from "@apollo/client";
+
+//Material UI dependencies
+import {
+  Card,
+  CardContent,
+  AppBar,
+  Box,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+
+//Local dependencies
+import List from "../List";
+import { GET_CONTAINERS, GET_PR } from "../../gql/query";
+import "./style.css";
+
+const Container = () => {
+  const { data: containerData } = useQuery(GET_CONTAINERS); // Fetching all 4 of our containers from graphQL
+  const { loading, data: prData } = useQuery(GET_PR); //Fetching all of the PRs and loading state
+
+  const [containers, setContainers] = useState(); //Using this state to populate the containers.
+  const [opened, setOpened] = useState(); //Using this state to store 'open' status PRs
+  const [closed, setClosed] = useState(); //Using this state to store 'closed' status PRs
+  const [assigned, setAssigned] = useState(); //Using this state to store prs that have someone assigned to it.
+  const [draft, setDraft] = useState(); //Using this state to store the draft PRs
+  useEffect(() => {
+    if (containerData) {
+      setContainers(containerData.containers); //Setting up the container state
+    }
+    if (prData) {
+      const pullRequests = prData.pullRequests.map((item) => item.github);
```

Comment:
these should all be outside the useEffect and used directly without storing them in state.

```js
const pullRequests = React.useMemo(() => prData?.pullRequests?.map(item => item.github)?.[0] || [], [prData?.pullRequests])

const opened = React.useMemo(() => pullRequests[0].filter((item) => item.state === "open"), [pullRequests])
...
```